src/core/dbus-service.c

```diff
@@ -143,6 +143,29 @@ static int bus_service_set_transient_property(
 
                 return 1;
 
+        } else if (streq(name, "Restart")) {
```

Comment:
well, "sytemd-run" won't reestablish the tty on HUP... try something without "-t"...